dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java

```diff
@@ -182,36 +182,29 @@ public DotTempFile createTempFile(final String incomingFileName,final HttpServle
    * @return
    * @throws DotSecurityException
    */
-  public DotTempFile createTempFileFromUrl(final String incomingFileName,final HttpServletRequest request, final URL url, final int timeoutSeconds, final long maxLength)
-      throws DotSecurityException {
-
-    if (!validUrl(url)) {
-      throw new DotSecurityException("Invalid url attempted for tempFile : " + url);
-    }
-
+  public DotTempFile createTempFileFromUrl(final String incomingFileName,
+          final HttpServletRequest request, final URL url, final int timeoutSeconds,
+          final long maxLength)
+          throws DotSecurityException, IOException {
 
+      if (!validUrl(url)) {
+          throw new BadRequestException("Invalid url attempted for tempFile : " + url);
```

Comment:
It is ok to throw this exception at this level?

Reply:
done, moved to Resource